### DISCUSSION TITLE
PippyG & PiPistrelle

### DISCUSSION THREAD

**Discussion by blogdemoi** [rdz-oss/BattyBirdNET-Pi#33] [General]
Hi,

I may be interested on setting up a BattyBirdNET-Pi (I already have a Birdnet-Pi system) and found out these microphones:
PippyG & PiPistrelle https://shop.smithrobotics.co.uk/ which might be of interest...

**Comment by rdz-oss** [comment]
Hi @blogdemoi  , 
those do look promising. The currently do not ship outside of the UK so I have not tested this combination yet. But @Pete-the-Geek  is alos looking into that option!

Most likely that should work out of the box ( as long as can be configured as a USB Microphone). There will, however, also be some initial issues with new types of noise that can be misinterpreted as bats. That would not be uncommon and can be fixed if we get our hands on about 200 samples of misclassified noise from this microphone.

Perhaps @Pete-the-Geek has some insights?

Best,
Richard

**Comment by Pete-the-Geek** [comment]
I am indeed planning on looking in to that. I now have the Pi Pipistrelles. I need to configure 1 to be a USB microphone and will set up my dev RPi4 with a clean install and see how it works. 

As for the Pi Pipistrelles, they seem to be very good and exceptionally well made by Smiths Robotics.
They are in now way environmentally sealed and nir can they. In my tests, I've been deploying them in a ziplock b bag. This wouldn't work for a permanent solution. They have their own noise profiles but no worse than the AudioMoth from my early tests. Time will tell...

**Comment by Pete-the-Geek** [comment]
Technical difficulties with turning the PI.Pipistrelle into a USB mic has temporarily prevented this. I've raid it to Phil, the developer so hopefully he can help. I'll then be able to continue testing. Watch this space (still)

**Comment by Pete-the-Geek** [comment]
So... after a lot of work, I have got it to work...

I had to put the Pi Pipistrelle into USB Mic mode by simply writing a UF2 file into the bootloader "partition" of the device. It then automatically reboots into the correct mode. To switch it back, you simply do the same procedure but with the "standard" firmware. This is similar to the audiomoth.

Installing the USB Mic for BattyBirdNET-Pi was interesting. I did the install over and over again and all I could do was get it to work from 0 Hz to 25 kHz. Anything above this recorded nothing. arecord was able to record the full 384 kHz but SOX rec only managed 25 kHz.
Resolved finally by changing the audio settings in raspi-config to use pulseaudio before installing BattyBirdNet-Pi and then it all worked correctly.  There may be an easier way but this was the only method I managed to make it work with.

They are quite noisy. This is recorded at home (so there is some minor interference)
![image](https://github.com/user-attachments/assets/7ced4b82-761e-466c-8b1b-672e13d81ffd)
Similarly, this is a recording from the same position with the EM2 Pro
![NYCNOC_20240715_220713 wav](https://github.com/user-attachments/assets/5b940976-933e-423c-98e6-fe9db16339b6)

And this is when put in overnight recording mode. Most of the extra noise seems to be in the writing to the SD card I think. You can actually hear it squeal
![image](https://github.com/user-attachments/assets/ea95180f-877c-44b8-9064-875f9aeaeb17)
I did create a noise profile for this which then looks like this
![image](https://github.com/user-attachments/assets/863c231e-3fa0-4db7-86e5-d5afa89464f5)

By no means perfect but for £40 it is a superb bit of kit.
It acts as a Heterodyne and Full Spectrum (Time Expansion x10 or x16) realtime/batwalk detector in one mode with manual one file record option. It does not  record and play sounds at the same time. It is an advanced heterodyne scanner though that auto scans across the frequency spectrum and locks on to a frequency when it detects a bat. Lights display the detected frequency as well as an optional speech synthesis of the frequency.
It acts as a USB Microphone though it is not supported by many applications currently
It acts as a passive bat detector where is can sleep/wake to record on triggered event.

See this page for all of the details https://www.omenie.com/pipistrelle.html I'm not sure how his recordings are so clean, I'm working on that...

**Reply by rdz-oss** [reply]
That is great work Pete! 
Great you managed to get the full spectrum recognized. I wonder why it has trouble with that on RaspberryOS. 
The noise patterns look like it includes some harmonics to me. Something that starts around 10kHz.
Perhaps the developer of the device has some insight on this?

**Reply by Pete-the-Geek** [reply]
Hoodlum he will have. I've sent my findings to him and asked what may be causing it and what I may be able to do to reduce it. I might take it away to a known quiet spot and get it to do some recordings. I know the RPi4 is very noisy so I hid the mic outside and left the Pi inside with a thick wall separating it. More testing is probably needed and some advice from the developer.

**Reply by rdz-oss** [reply]
I was at the Smith Engineering Booth at the BCT convention and was able to test the brand new new usb version as well as the pippyg directly on the system. The USB version looks promissing. Still has some horizontal ines/noise but better. I also recorded 4 bat species on a batwalk using the pippyg as microphone. Working alright - some adjustments to the classifier needs to be done though. And a bit noisy ( as in your recorings). But works in general.

**Reply by Pete-the-Geek** [reply]
That's great. It must have really helped that you could get it out live!!!

I do hope they can clear up the noise but I may have to buy a new built one once they figure out the recipe.

**Reply by rdz-oss** [reply]
They have a new usb version without a batterypack that is not yet available. That one was a bit better. A prototype at the booth. That one might be interesting for BattyBirdNET-Pi systems - especially once they also offer a cover. That is on the roadmap apparently. I just saw its on the website already as new usb version https://www.omenie.com/pipistrelle.html
"NO HAND SOLDERING - EXTREMELY SIMPLE BUILD​ IN UNDER 10 MINUTES, APPROX £10 PER DEVICE WHEN BUILDING 10" That is quite a low cost.

**Reply by MueJosh** [reply]
> They have a new usb version without a batterypack that is not yet available. That one was a bit better. A prototype at the booth. That one might be interesting for BattyBirdNET-Pi systems - especially once they also offer a cover. That is on the roadmap apparently. I just saw its on the website already as new usb version https://www.omenie.com/pipistrelle.html
> 
> "NO HAND SOLDERING - EXTREMELY SIMPLE BUILD​ IN UNDER 10 MINUTES, APPROX £10 PER DEVICE WHEN BUILDING 10" That is quite a low cost.
> 
> 

I think that's the one I have, I've send you an email.

**Comment by rabede** [comment]
Pipistrelle Mini and Pippyg  are now available in EU from here:
https://www.apodemus.eu/de/Apodemus-Pipistrelle-Mini/00126.2 
https://www.apodemus.eu/de/Apodemus-Pippyg/00127.1

**Comment by Fish31520** [comment]
@rdz-oss @Pete-the-Geek @blogdemoi 
Hi,

I just bought one of those Pippyg. Do you guys know how to change the firmware so it can act as a USB mic for BattyBirdNet ?

Thanks a lot !

Jean

[Edit : My bad, I managed to change the firmware, its now fully functional as a microphone, but it seems to work as a regular mic, not a bat/ultrasounds microphone...

Should I change the "Microphone noise profile", it's currently on Audiomoth_v12 ?]
